### PR TITLE
SI-8092 Try to verify f-interpolator format

### PR DESCRIPTION
**Pull request by som-snytt**
There is a private parse routine on Formatter in OpenJDK,
so use it if it's there to verify the format string.

Otherwise, try performing the format with dummy arguments.

### CODE REVIEW

File: src/compiler/scala/tools/reflect/MacroImplementations.scala [hunk 1]
```diff
@@ -155,6 +155,59 @@ abstract class MacroImplementations {
...
+        true
+      }
+      /** Attempt to verify the format string by invoking
+       *  the private `parse` method, available on OpenJDK.
+       */
+      def attemptFormatParse(s: String): Boolean = {
+        import scala.reflect.runtime._
+        import universe._
+        val im = currentMirror reflect (new Formatter)
+        typeOf[Formatter] member TermName("parse") match {
```
Comment (retronym) [inline src/compiler/scala/tools/reflect/MacroImplementations.scala]: We don't want the compiler to use Scala runtime reflection. It has a big startup cost, and it loads classes that might make hot calls in the compiler itself megamorphic.

**Comment by retronym** [comment]
Can you provide some examples of f-interpoolations that are currently checkable vs non-checkable, and explain where the boundary lies?

File: src/compiler/scala/tools/reflect/MacroImplementations.scala (hunk 1 above)
Comment (som-snytt) [inline src/compiler/scala/tools/reflect/MacroImplementations.scala]: Agreed. That represents my frustration that they didn't expose a parse method.

**Comment by som-snytt** [comment]
The test example is the one I hit; I thought of this PR as a back-stop, but maybe it's worth the maintenance to just complete the macro's format parsing.

**Comment by som-snytt** [comment]
The test fail was double-reporting a bad conversion.

**Comment by som-snytt** [comment]
Another example, wondering if G means Greenwich Mean:

```
scala> f"$now%tG"
java.util.UnknownFormatConversionException: Conversion = 'tG'
```

Anyway, invoking internal methods, even as an optional strategy, has got to be bad policy.

**Comment by som-snytt** [comment]
The back-stop would have to handle a few more types (Formattable, BigInt) to be useful, and also use the target type the arg is eval'd to.

**Pull request closed without merging**